### PR TITLE
[feature](iceberg)support iceberg insert grammar

### PR DESCRIPTION
**Pull request by wuwenchi**
## Proposed changes

Issue #31442

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wuwenchi** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/HiveTableSink.java [hunk 1]
```diff
@@ -50,16 +49,25 @@
...
-public class HiveTableSink extends DataSink {
+public class HiveTableSink extends BaseExternalTableDataSink {
 
-    private HMSExternalTable targetTable;
-    protected TDataSink tDataSink;
+    private final HMSExternalTable targetTable;
+    private static final HashSet<TFileFormatType> supportedTypes = new HashSet<TFileFormatType>() {{
+            add(TFileFormatType.FORMAT_ORC);
+            add(TFileFormatType.FORMAT_PARQUET);
+            add(TFileFormatType.FORMAT_CSV_PLAIN);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/HiveTableSink.java:58]: FORMAT_CSV_PLAIN is not supported yet

File: fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundTableSinkCreator.java [hunk 2]
```diff
@@ -72,6 +76,9 @@ public static LogicalSink<? extends Plan> createUnboundTableSink(List<String> na
         } else if (curCatalog instanceof HMSExternalCatalog) {
             return new UnboundHiveTableSink<>(nameParts, colNames, hints, partitions,
                     dmlCommandType, Optional.empty(), Optional.empty(), plan);
+        } else if (curCatalog instanceof IcebergExternalCatalog) {
```
[the remark this answers is not in the document]
Comment (wuwenchi) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundTableSinkCreator.java:79]: This check will be done when it comes to logic to physics:
```
    private Pair<HMSExternalDatabase, HMSExternalTable> bind(CascadesContext cascadesContext,
                                                             UnboundHiveTableSink<? extends Plan> sink) {
        List<String> tableQualifier = RelationUtil.getQualifierName(cascadesContext.getConnectContext(),
                sink.getNameParts());
        Pair<DatabaseIf<?>, TableIf> pair = RelationUtil.getDbAndTable(tableQualifier,
                cascadesContext.getConnectContext().getEnv());
        if (pair.second instanceof HMSExternalTable) {
            return Pair.of(((HMSExternalDatabase) pair.first), (HMSExternalTable) pair.second);
        }
        throw new AnalysisException("the target table of insert into is not a Hive table");
    
```

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by kaka11chen — APPROVED** [review]
LGTM

**Pull request merged**